File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string code) {
    string answer = "";
    string ret;
    
    bool mode = 0;

    
    
    int idx = 0;
    while (idx < code.size())
    {
        if (code[idx] == '1')
        {
            mode = !mode;
            idx++;
            continue;
        }
        
        if (mode == 0 && idx % 2 == 0)
        {
            ret += code[idx];
        }
        else if (mode == 1 && idx % 2 == 1)
        {
            ret += code[idx];
        }
        
        idx++;
    }
    if (ret[0] == '\0')
        return "EMPTY";
    answer = ret;
    return answer;
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    
    ret = ''
    
    mode = False

    
    
    idx = 0
    while idx < len(code):
    
        if code[idx] == '1':
            mode = not mode
            idx+= 1
            continue
        
        
        if mode == 0 and idx % 2 == 0:
            ret += code[idx]
        
        elif mode == 1 and idx % 2 == 1:
            ret += code[idx]
        
        idx+=1
    
    if not ret: 
        return "EMPTY"
    
    answer = ret
    
    
    return answer